acme/dns_challenge.go

```diff
@@ -85,14 +85,22 @@ func (s *dnsChallenge) Solve(chlng challenge, domain string) error {
 		return err
 	}
 
-	err = s.provider.Present(domain, chlng.Token, keyAuth)
-	if err != nil {
-		return fmt.Errorf("Error presenting token: %s", err)
+	for _, v := range s.providers {
+		err = v.Present(domain, chlng.Token, keyAuth)
```

Comment:
This would try to `Present()` every domain on every provider. Shouldn't we have a way to tell which provider a certain domain uses?